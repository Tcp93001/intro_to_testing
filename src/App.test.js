// import React from 'react';
// import { render } from '@testing-library/react';
// import renderer from 'react-test-renderer'
// import App from './App';

// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// it('render the element vs snapshot', () =>{
//   const component = renderer.create(<App />)
//   let tree = component.toJSON()
//   expect(tree).toMatchSnapshot()
// })

// it('render a <p> element', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/Edit/i);
//   expect(linkElement).toBeInTheDocument();
// })

// test('renderea un texto Edit', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/edit/i);
//   expect(linkElement).toBeInTheDocument();
// });

import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import App from './App'

describe('Mejor App testing', () => {

  const component = mount(<App />)
  
  it('should render an Edit', () => {
    const wrapper = shallow(<App />)
    const parrafo = wrapper.find('p')
    const result = parrafo.at(1).text()
    // const result = parrafo.text()
    // console.log(result)
    // expect(result).toBe("")
    // expect(result).toEqual(expect.stringContaining('Edit'))
    // expect(result).toEqual(expect.stringContaining('Edit'))
  })

  it('should render an Edit better test', () => {
    // expect(component.find('p').at(0).text()).toEqual(expect.stringContaining('Ejemplo'))
    expect(component.find('p').at(0).text()).toMatch(/Ejemplo/)
  })

  it('render the element vs snapshot', () =>{
    // const component = mount(<App />)
    // let tree = component.toJSON()
    // expect(tree).toMatchSnapshot()
    expect(component).toMatchSnapshot()
  })
})

