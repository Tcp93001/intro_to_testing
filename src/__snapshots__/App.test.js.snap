// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mejor App testing render the element vs snapshot 1`] = `ReactWrapper {}`;
